Detections/SecurityEvent/UserAccountAdd-Removed.yaml

```diff
@@ -25,26 +25,61 @@ query: |
   let WellKnownLocalSID = "S-1-5-32-5[0-9][0-9]$";
   let WellKnownGroupSID = "S-1-5-21-[0-9]*-[0-9]*-[0-9]*-5[0-9][0-9]$|S-1-5-21-[0-9]*-[0-9]*-[0-9]*-1102$|S-1-5-21-[0-9]*-[0-9]*-[0-9]*-1103$|S-1-5-21-[0-9]*-[0-9]*-[0-9]*-498$|S-1-5-21-[0-9]*-[0-9]*-[0-9]*-1000$";
   let AC_Add = 
-  SecurityEvent
+  (union isfuzzy=true 
+  (SecurityEvent
   // Event ID related to member addition.
   | where EventID in (4728, 4732,4756) 
   | where TargetSid matches regex WellKnownLocalSID or TargetSid matches regex WellKnownGroupSID  
   | parse EventData with * '"MemberName">' * '=' AccountAdded ",OU" *
   | where isnotempty(AccountAdded)
   | extend GroupAddedTo = TargetUserName, AddingAccount = Account 
   | extend  AccountAdded_GroupAddedTo_AddingAccount = strcat(AccountAdded, "||", GroupAddedTo, "||", AddingAccount )
-  | project AccountAdded_GroupAddedTo_AddingAccount, AccountAddedTime = TimeGenerated;
+  | project AccountAdded_GroupAddedTo_AddingAccount, AccountAddedTime = TimeGenerated
+  ),
+  (WindowsEvent
+  // Event ID related to member addition.
+  | where EventID in (4728, 4732,4756) 
+  | extend TargetSid = tostring(EventData.TargetSid)
+  | where TargetSid matches regex WellKnownLocalSID or TargetSid matches regex WellKnownGroupSID  
+  | parse EventData with * '"MemberName">' * '=' AccountAdded ",OU" *
```

Comment:
First, I would parse the relevant field, and not EventData as a whole. Secondly, there might be a bug here. Per [this](https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4732) the field does not have ,OU in it.

Reply:
I changed the parsing, what should I do with the potential bug?

Reply:
I will ask MSTIC